### ISSUE TITLE
"Unknown action or rule field" error when declaring a rule with enabled_if and without target

### ISSUE THREAD

**Issue by jchavarri** [ocaml/dune#3885]
## Expected Behavior

When adding rule without `target` and with `enabled_if` like:

```
(rule
 (enabled_if
  (= "re" %{env:ODOC_SYNTAX=ml}))
 (action (copy my_page.mld.re my_page.mld)))
```

It should error with message "`target` is missing in `rule` stanza".

## Actual Behavior

It errors with confusing message:

```
File "lib/dune", line 12, characters 2-12:
12 |  (enabled_if
       ^^^^^^^^^^
Error: Unknown action or rule field.
```

Side note: I've noticed the same error is returned if the `rule` has target but coming after `enabled_if`, e.g.:

```
(rule
 (enabled_if
  (= "re" %{env:ODOC_SYNTAX=ml}))
 (target my_page.mld)
 (action (copy my_page.mld.re my_page.mld)))
```

Not sure if this is expected?

## Specifications

- Version of `dune` (output of `dune --version`): 2.7.0
- Version of `ocaml` (output of `ocamlc --version`): 4.10.0
- Operating system (distribution and version): macOS 10.15.7

**Comment by nojb** [comment]
This has been fixed in #3831

**Comment by nojb** [comment]
I'm going to close this one, as the problem has been fixed. Thanks!
